Example/Example/AppFlowController.swift

```diff
@@ -17,7 +17,7 @@ final class AppFlowController: UIViewController {
     super.init(nibName: nil, bundle: nil)
 
     let purchaseLogicController = PurchaseLogicController(
-      checkoutURLProvider: getCheckoutURL,
+      checkoutURLProvider: Repository.shared.checkout(email:amount:completion:),
```

Comment:
If I was to fix up `.shared` I think I would make `Dependencies` a `struct` where the work performed in `initializeDependencies` moves into the initializer and `repository` is a property.

`AppFlowController` would probably receive the entire `Dependencies` `struct`

This is probably fine for now though